### PR TITLE
Reorder sub-function placement

### PR DESCRIPTION
**Pull request by ender503**
To eliminate the sub-function used before it declared, put the
sub-function to the front.

### CODE REVIEW

File: utils/cert.js [hunk 1]
```diff
@@ -2,6 +2,51 @@ const crypto = require('crypto');
 const tools = require('./tools.js');
 const iota = require('./iotaSetup.js');
 
+// IOTA - Attach your transfer to the tangle
```
Comment (jserv) [inline utils/cert.js]: s/your transfer/a transaction/

File: utils/cert.js [hunk 2]
```diff
@@ -2,6 +2,51 @@ const crypto = require('crypto');
...
+  const p = new Promise((res, rej) => {
+    iota.api.findTransactions(query, (e, s) => {
+      if (e) return rej(e);
+      res(s);
+    });
+  });
+  return p;
+};
+
+// IOTA - Attach your transfer to the tangle
```
Comment (jserv) [inline utils/cert.js]: Is the comment perfectly describing the following lines?

File: utils/cert.js [hunk 3]
```diff
@@ -2,6 +2,51 @@ const crypto = require('crypto');
...
+    console.log('Attaching to Tangle');
+    iota.api.sendTransfer(seed, 6, 15, transfers, (e, s) => {
+      if (e) return rej(e);
+      res(s);
+    });
+  });
+  return p;
+};
+
+// IOTA - Attach your transfer to the tangle
```
Comment (jserv) [inline utils/cert.js]: Ditto. Revise comment(s).

File: utils/tools.js [hunk 4]
```diff
@@ -1,6 +1,14 @@
 const forge = require('node-forge');
 const crypto = require('crypto');
 
+const d64 = data => forge.util.decode64(data);
```
Comment (jserv) [inline utils/tools.js]: Avoid unnecessary blank lines.

**Comment by jserv** [comment]
You shall mention style check for such modifications as well.

**Comment by jserv** [comment]
It is not "position" but "placement".

**Comment by jserv** [comment]
Declare all sub-functions before its actual use.  Because this gets rid of the risk of using a random value by mistake. In addition, this can give optimization hints to interpreters or JIT compilers. Separating declaration and initialization also prevents you from using "const" (or "final") when you could.

File: utils/tools.js (hunk 4 above)
Comment (ender503) [reply utils/tools.js]: These const variables are representing as the function, should we remove the blank line before these functions? 🤔

Comment (jserv) [reply utils/tools.js]: I think so.

File: utils/cert.js (hunk 2 above)
Comment (ender503) [reply utils/cert.js]: Code documentation updated.

File: utils/tools.js (hunk 4 above)
Comment (ender503) [reply utils/tools.js]: Unnecessary blank lines are removed. 👍

**Pull request merged**